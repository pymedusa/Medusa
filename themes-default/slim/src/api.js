import axios from 'axios';

// This should be more dynamic. As now when we change the apiKey in config/general. This won't work anymore.
// Because of this, a page reload is required.
export const webRoot = document.body.getAttribute('web-root');
export const apiKey = document.body.getAttribute('api-key');

/**
 * Api client based on the axios client, to communicate with medusa's web routes, which return json data.
 */
export const apiRoute = axios.create({
    baseURL: webRoot + '/',
    timeout: 60000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
    }
});

/**
 * Api client based on the axios client, to communicate with medusa's api v1.
 */
export const apiv1 = axios.create({
    baseURL: webRoot + '/api/v1/' + apiKey + '/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
    }
});

/**
 * Api client based on the axios client, to communicate with medusa's api v2.
 */
export const api = axios.create({
    baseURL: webRoot + '/api/v2/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        'X-Api-Key': apiKey
    }
});
