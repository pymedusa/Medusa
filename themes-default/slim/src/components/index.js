export { default as AddRecommended } from './add-recommended.vue';
export { default as AddShowOptions } from './add-show-options.vue';
export { default as AddShows } from './add-shows.vue';
export { default as AnidbReleaseGroupUi } from './anidb-release-group-ui.vue';
export { default as AppFooter } from './app-footer.vue';
export { default as AppHeader } from './app-header.vue';
export { default as Backstretch } from './backstretch.vue';
export { default as Config } from './config.vue';
export { default as ConfigGeneral } from './config-general.vue';
export { default as ConfigPostProcessing } from './config-post-processing.vue';
export { default as ConfigNotifications } from './config-notifications.vue';
export { default as ConfigSearch } from './config-search.vue';
export { default as DisplayShow } from './display-show.vue';
export { default as EditShow } from './edit-show.vue';
export { default as History } from './history.vue';
export { default as Home } from './home.vue';
export { default as IRC } from './irc.vue';
export { default as Login } from './login.vue';
export { default as Logs } from './logs.vue';
export { default as ManualPostProcess } from './manual-post-process.vue';
export { default as RootDirs } from './root-dirs.vue';
export { default as Schedule } from './schedule.vue';
export { default as ShowHeader } from './show-header.vue';
export { default as ShowHistory } from './show-history.vue';
export { default as ShowResults } from './show-results.vue';
export { default as SnatchSelection } from './snatch-selection.vue';
export { default as Status } from './status.vue';
export { default as SubMenu } from './sub-menu.vue';
export { default as SubtitleSearch } from './subtitle-search.vue';
export * from './http';
export * from './helpers';
