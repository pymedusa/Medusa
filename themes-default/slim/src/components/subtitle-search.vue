<template>
    <tr class="subtitle-search-wrapper">
        <td colspan="9999">
            <span v-if="loading" class="loading-message">{{loadingMessage}} <state-switch :theme="config.themeName" state="loading" /></span>
            <div v-if="displayQuestion" class="search-question">
                <div class="question">
                    <p v-if="!lang">Do you want to manually pick subtitles or let us choose it for you?</p>
                    <p v-else>Do you want to manually pick subtitles or search a subtitle with the language code <b>{{lang}}</b> for you?</p>
                </div>
                <div class="options">
                    <button type="button" class="btn-medusa btn-info" @click="autoSearch">Auto</button>
                    <button type="button" class="btn-medusa btn-success" @click="manualSearch">Manual</button>
                </div>
            </div>

            <div v-if="subtitles.length > 0" class="subtitle-results">
                <span class="release-name">
                    {{releaseName}}
                </span>
                <vue-good-table
                    :columns="columns"
                    :rows="subtitles"
                    :search-options="{
                        enabled: false
                    }"
                    :sort-options="{
                        enabled: true,
                        initialSortBy: { field: 'score', type: 'desc' }
                    }"
                    styleClass="vgt-table condensed subtitle-table"
                >
                    <template v-slot:table-column="props">
                        <span v-if="props.column.label === 'Download'">
                            <span>{{props.column.label}}</span>
                            <span class="btn-medusa btn-xs pull-right" @click="close">hide</span>
                        </span>
                        <span v-else>
                            {{props.column.label}}
                        </span>
                    </template>
                    <template v-slot:table-row="props">
                        <span v-if="props.column.field === 'provider'">
                            <img :src="`images/subtitles/${props.row.provider}.png`" width="16" height="16">
                            <span :title="props.row.provider">{{props.row.provider}}</span>
                        </span>
                        <span v-else-if="props.column.field === 'lang'">
                            <img :title="props.row.lang" :src="`images/subtitles/flags/${props.row.lang}.png`" width="16" height="11">
                        </span>
                        <span v-else-if="props.column.field === 'filename'">
                            <a :title="`Download${props.row.hearing_impaired ? ' hearing impaired ' : ' '}subtitle: ${props.row.filename}`" @click="pickSubtitle(props.row.id)">
                                <img v-if="props.row.hearing_impaired" src="images/hearing_impaired.png" width="16" height="16">
                                <span class="subtitle-name">{{props.row.filename}}</span>
                                <img v-if="props.row.sub_score >= props.row.min_score" src="images/save.png" width="16" height="16">
                            </a>
                        </span>
                        <span v-else-if="props.column.field === 'download'">
                            <a :title="`Download${props.row.hearing_impaired ? ' hearing impaired ' : ' '}subtitle: ${props.row.filename}`" @click="pickSubtitle(props.row.id)">
                                <img src="images/download.png" width="16" height="16">
                            </a>
                        </span>
                        <span v-else>
                            {{props.formattedRow[props.column.field]}}
                        </span>
                    </template>
                </vue-good-table>
            </div>

        </td>
    </tr>
</template>
<script>

import { mapActions, mapGetters, mapState } from 'vuex';
import { VueGoodTable } from 'vue-good-table';
import { apiRoute } from '../api';
import { StateSwitch } from './helpers';

export default {
    name: 'subtitle-search',
    components: {
        StateSwitch,
        VueGoodTable
    },
    props: {
        show: {
            type: Object,
            required: true
        },
        episode: {
            type: Object,
            required: true
        },
        lang: {
            type: String,
            required: false
        }
    },
    data() {
        return {
            columns: [{
                label: 'Filename',
                field: 'filename'
            }, {
                label: 'Language',
                field: 'lang'
            }, {
                label: 'Provider',
                field: 'provider'
            }, {
                label: 'Score',
                field: 'score',
                type: 'number'
            }, {
                label: 'Sub Score',
                field: 'sub_score',
                type: 'number'
            }, {
                label: 'Missing Matches',
                field: rowObj => {
                    if (rowObj.missing_guess) {
                        return rowObj.missing_guess.join(', ');
                    }
                },
                type: 'array'
            }, {
                label: 'Download',
                field: 'download',
                sortable: false
            }],
            subtitles: [],
            displayQuestion: false,
            loading: false,
            loadingMessage: ''
        };
    },
    computed: {
        ...mapState({
            config: state => state.config
        }),
        ...mapGetters({
            getLastReleaseName: 'getLastReleaseName'
        }),
        releaseName() {
            const { episode, getLastReleaseName, show } = this;
            const lastKnownReleaseName = getLastReleaseName({ showSlug: show.id.slug, episodeSlug: episode.slug });
            if (lastKnownReleaseName) {
                return `Select subtitle for release: ${lastKnownReleaseName}`;
            }
            return 'Could not get a downloaded release name from the history table';
        },
        subtitleParams() {
            const { episode, show } = this;
            const params = {
                indexername: show.indexer,
                seriesid: show.id[show.indexer],
                season: episode.season,
                episode: episode.episode
            };

            return params;
        }
    },
    mounted() {
        this.displayQuestion = true;
    },
    methods: {
        ...mapActions([
            'getShowEpisodeHistory'
        ]),
        autoSearch() {
            const { lang, subtitleParams } = this;
            const params = subtitleParams;

            if (lang !== undefined) {
                params.lang = lang;
            }

            this.displayQuestion = false;
            this.loadingMessage = 'Searching for subtitles and downloading if available... ';
            this.loading = true;
            apiRoute('home/searchEpisodeSubtitles', { params })
                .then(response => {
                    if (response.data.result !== 'failure') {
                        // Update the show, as we have new information (subtitles)
                        // Let's emit an event, telling the displayShow component, to update the show using the api/store.
                        this.$emit('update', {
                            reason: 'new subtitles found',
                            codes: response.data.subtitles,
                            languages: response.data.languages
                        });
                    }
                })
                .catch(error => {
                    console.log(`Error trying to search for subtitles. Error: ${error}`);
                })
                .finally(() => {
                    // Cleanup
                    this.loadingMessage = '';
                    this.loading = false;
                    this.close();
                });
        },
        manualSearch() {
            const { episode, getShowEpisodeHistory, show, subtitleParams } = this;

            this.displayQuestion = false;
            this.loading = true;
            this.loadingMessage = 'Searching for subtitles... ';
            apiRoute('home/manualSearchSubtitles', { params: subtitleParams })
                .then(response => {
                    if (response.data.result === 'success') {
                        this.subtitles.push(...response.data.subtitles);
                        // Load episode history if we have subtitles.
                        getShowEpisodeHistory({ showSlug: show.id.slug, episodeSlug: episode.slug });
                    }
                }).catch(error => {
                    console.log(`Error trying to search for subtitles. Error: ${error}`);
                }).finally(() => {
                    this.loading = false;
                });
        },
        redownloadLang() {
            const { subtitleParams } = this;

            apiRoute('home/searchEpisodeSubtitles', { params: subtitleParams })
                .then(response => {
                    if (response.data.result !== 'failure') {
                        // Update the show, as we have new information (subtitles)
                        // Let's emit an event, telling the displayShow component, to update the show using the api/store.
                        this.$emit('update', {
                            reason: 'new subtitles found',
                            codes: response.data.subtitles,
                            languages: response.data.languages
                        });
                    }
                })
                .catch(error => {
                    console.log(`Error trying to search for subtitles. Error: ${error}`);
                })
                .finally(() => {
                    // Cleanup
                    this.loadingMessage = '';
                    this.loading = false;
                    this.close();
                });
        },
        pickSubtitle(subtitleId) {
            // Download and save this subtitle with the episode.
            const { subtitleParams } = this;
            const params = {
                ...subtitleParams,
                picked_id: subtitleId // eslint-disable-line camelcase
            };

            this.displayQuestion = false;
            this.loadingMessage = 'downloading subtitle... ';
            this.loading = true;

            apiRoute('home/manualSearchSubtitles', { params })
                .then(response => {
                    if (response.data.result === 'success') {
                        // Update the show, as we have new information (subtitles)
                        // Let's emit an event, telling the displayShow component, to update the show using the api/store.
                        this.$emit('update', {
                            reason: 'new subtitles found',
                            codes: response.data.subtitles,
                            languages: response.data.languages
                        });
                    }
                })
                .catch(error => {
                    console.log(`Error trying to search for subtitles. Error: ${error}`);
                })
                .finally(() => {
                    // Cleanup
                    this.loadingMessage = '';
                    this.loading = false;
                    this.close();
                });
        },
        close() {
            this.$emit('close');
            // Destroy the vue listeners, etc
            this.$destroy();
            // Remove the element from the DOM
            this.$el.remove();
        }
    }
};
</script>
<style scoped>
.subtitle-search-wrapper {
    display: table-row;
    column-span: all;
}

.subtitle-search-wrapper >>> table.subtitle-table tr {
    background-color: rgb(190, 222, 237);
}

.subtitle-search-wrapper > td {
    padding: 0;
}

.search-question,
.loading-message {
    background-color: rgb(51, 51, 51);
    color: rgb(255, 255, 255);
    padding: 10px;
    line-height: 55px;
}

span.subtitle-name {
    color: rgb(0, 0, 0);
}

span.release-name {
    text-align: left;
    width: 100%;
    display: block;
    background-color: rgb(51, 51, 51);
    color: rgb(255, 255, 255);
    padding: 4px;
}
</style>
