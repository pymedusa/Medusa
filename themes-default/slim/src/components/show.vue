<script>
import { mapState, mapGetters, mapActions } from 'vuex';
import { apiRoute } from '../api';
import { AppLink, PlotInfo } from './helpers';
import ShowHeader from './show-header.vue';

export default {
    name: 'show',
    template: '#show-template',
    components: {
        AppLink,
        PlotInfo,
        ShowHeader
    },
    metaInfo() {
        if (!this.show || !this.show.title) {
            return {
                title: 'Medusa'
            };
        }
        const { title } = this.show;
        return {
            title,
            titleTemplate: '%s | Medusa'
        };
    },
    props: {
        /**
         * Show indexer
         */
        showIndexer: {
            type: String
        },
        /**
         * Show id
         */
        showId: {
            type: Number
        }
    },
    data() {
        return {};
    },
    computed: {
        ...mapState({
            shows: state => state.shows.shows
        }),
        ...mapGetters({
            getShowById: 'getShowById',
            show: 'getCurrentShow'
        }),
        indexer() {
            return this.showIndexer || this.$route.query.indexername;
        },
        id() {
            return this.showId || Number(this.$route.query.seriesid) || undefined;
        }
    },
    mounted() {
        const {
            id,
            indexer,
            getShow,
            setEpisodeSceneNumbering,
            setAbsoluteSceneNumbering,
            setInputValidInvalid,
            getSeasonSceneExceptions,
            $store,
            show
        } = this;

        // Let's tell the store which show we currently want as current.
        $store.commit('currentShow', {
            indexer,
            id
        });

        // We need the episodes, so let's get it.
        // Enable detailed as we need the xemNumbering for the show-header.
        if (!show || !show.seasons) {
            getShow({ id, indexer, detailed: true, episodes: true });
        }

        this.$watch('show', () => {
            this.$nextTick(() => this.reflowLayout());
        });

        ['load', 'resize'].map(event => {
            return window.addEventListener(event, () => {
                this.reflowLayout();
            });
        });

        window.addEventListener('load', () => {
            $.ajaxEpSearch({
                colorRow: true
            });

            startAjaxEpisodeSubtitles(); // eslint-disable-line no-undef
            $.ajaxEpSubtitlesSearch();
            $.ajaxEpRedownloadSubtitle();
        });

        $(document.body).on('click', '.seasonCheck', event => {
            const seasCheck = event.currentTarget;
            const seasNo = $(seasCheck).attr('id');

            $('#collapseSeason-' + seasNo).collapse('show');
            const seasonIdentifier = 's' + seasNo;
            $('.epCheck:visible').each((index, element) => {
                const epParts = $(element).attr('id').split('e');
                if (epParts[0] === seasonIdentifier) {
                    element.checked = seasCheck.checked;
                }
            });
        });

        let lastCheck = null;
        $(document.body).on('click', '.epCheck', event => {
            const target = event.currentTarget;
            if (!lastCheck || !event.shiftKey) {
                lastCheck = target;
                return;
            }

            const check = target;
            let found = 0;

            $('.epCheck').each((index, element) => {
                if (found === 1) {
                    element.checked = lastCheck.checked;
                }

                if (found === 2) {
                    return false;
                }

                if (element === check || element === lastCheck) {
                    found++;
                }
            });
        });

        // Initially show/hide all the rows according to the checkboxes
        $('#checkboxControls input').each((index, element) => {
            const status = $(element).prop('checked');
            $('tr.' + $(element).attr('id')).each((index, tableRow) => {
                if (status) {
                    $(tableRow).show();
                } else {
                    $(tableRow).hide();
                }
            });
        });

        $(document.body).on('change', '.sceneSeasonXEpisode', event => {
            const target = event.currentTarget;
            // Strip non-numeric characters
            const value = $(target).val();
            $(target).val(value.replace(/[^0-9xX]*/g, ''));
            const forSeason = $(target).attr('data-for-season');
            const forEpisode = $(target).attr('data-for-episode');

            // If empty reset the field
            if (value === '') {
                setEpisodeSceneNumbering(forSeason, forEpisode, null, null);
                return;
            }

            const m = $(target).val().match(/^(\d+)x(\d+)$/i);
            const onlyEpisode = $(target).val().match(/^(\d+)$/i);
            let sceneSeason = null;
            let sceneEpisode = null;
            let isValid = false;
            if (m) {
                sceneSeason = m[1];
                sceneEpisode = m[2];
                isValid = setInputValidInvalid(true, $(target));
            } else if (onlyEpisode) {
                // For example when '5' is filled in instead of '1x5', asume it's the first season
                sceneSeason = forSeason;
                sceneEpisode = onlyEpisode[1];
                isValid = setInputValidInvalid(true, $(target));
            } else {
                isValid = setInputValidInvalid(false, $(target));
            }

            if (isValid) {
                setEpisodeSceneNumbering(forSeason, forEpisode, sceneSeason, sceneEpisode);
            }
        });

        $(document.body).on('change', '.sceneAbsolute', event => {
            const target = event.currentTarget;
            // Strip non-numeric characters
            $(target).val($(target).val().replace(/[^0-9xX]*/g, ''));
            const forAbsolute = $(target).attr('data-for-absolute');

            const m = $(target).val().match(/^(\d{1,3})$/i);
            let sceneAbsolute = null;
            if (m) {
                sceneAbsolute = m[1];
            }
            setAbsoluteSceneNumbering(forAbsolute, sceneAbsolute);
        });

        $('#showTable, #animeTable').tablesorter({
            widgets: ['saveSort', 'stickyHeaders', 'columnSelector'],
            widgetOptions: {
                columnSelector_saveColumns: true, // eslint-disable-line camelcase
                columnSelector_layout: '<label><input type="checkbox">{name}</label>', // eslint-disable-line camelcase
                columnSelector_mediaquery: false, // eslint-disable-line camelcase
                columnSelector_cssChecked: 'checked' // eslint-disable-line camelcase
            }
        });

        $('#popover').popover({
            placement: 'bottom',
            html: true, // Required if content has HTML
            content: '<div id="popover-target"></div>'
        }).on('shown.bs.popover', () => { // Bootstrap popover event triggered when the popover opens
            $.tablesorter.columnSelector.attachTo($('#showTable, #animeTable'), '#popover-target');
        });

        // Changes the button when clicked for collapsing/expanding the season to show/hide episodes
        document.querySelectorAll('.collapse.toggle').forEach(element => {
            element.addEventListener('hide.bs.collapse', () => {
                // On hide
                const reg = /collapseSeason-(\d+)/g;
                const result = reg.exec(this.id);
                $('#showseason-' + result[1]).text('Show Episodes');
                $('#season-' + result[1] + '-cols').addClass('shadow');
            });
            element.addEventListener('show.bs.collapse', () => {
                // On show
                const reg = /collapseSeason-(\d+)/g;
                const result = reg.exec(this.id);
                $('#showseason-' + result[1]).text('Hide Episodes');
                $('#season-' + result[1] + '-cols').removeClass('shadow');
            });
        });

        // Get the season exceptions and the xem season mappings.
        getSeasonSceneExceptions();
    },
    methods: {
        ...mapActions({
            getShow: 'getShow' // Map `this.getShow()` to `this.$store.dispatch('getShow')`
        }),
        /**
         * Attaches IMDB tooltip,
         * Moves summary and checkbox controls backgrounds
         */
        reflowLayout() {
            console.debug('Reflowing layout');

            this.$nextTick(() => {
                this.moveSummaryBackground();
                this.movecheckboxControlsBackground();
            });

            attachImdbTooltip(); // eslint-disable-line no-undef
        },
        /**
         * Adjust the summary background position and size on page load and resize
         */
        moveSummaryBackground() {
            const height = $('#summary').height() + 10;
            const top = $('#summary').offset().top + 5;

            $('#summaryBackground').height(height);
            $('#summaryBackground').offset({ top, left: 0 });
            $('#summaryBackground').show();
        },
        /**
         * Adjust the checkbox controls (episode filter) background position
         */
        movecheckboxControlsBackground() {
            const height = $('#checkboxControls').height() + 10;
            const top = $('#checkboxControls').offset().top - 3;

            $('#checkboxControlsBackground').height(height);
            $('#checkboxControlsBackground').offset({ top, left: 0 });
            $('#checkboxControlsBackground').show();
        },
        setEpisodeSceneNumbering(forSeason, forEpisode, sceneSeason, sceneEpisode) {
            const indexerName = $('#indexer-name').val();
            const seriesId = $('#series-id').val();

            if (sceneSeason === '') {
                sceneSeason = null;
            }
            if (sceneEpisode === '') {
                sceneEpisode = null;
            }

            $.getJSON('home/setSceneNumbering', {
                indexername: indexerName,
                seriesid: seriesId,
                forSeason,
                forEpisode,
                sceneSeason,
                sceneEpisode
            }, data => {
                // Set the values we get back
                if (data.sceneSeason === null || data.sceneEpisode === null) {
                    $('#sceneSeasonXEpisode_' + seriesId + '_' + forSeason + '_' + forEpisode).val('');
                } else {
                    $('#sceneSeasonXEpisode_' + seriesId + '_' + forSeason + '_' + forEpisode).val(data.sceneSeason + 'x' + data.sceneEpisode);
                }
                if (!data.success) {
                    if (data.errorMessage) {
                        alert(data.errorMessage); // eslint-disable-line no-alert
                    } else {
                        alert('Update failed.'); // eslint-disable-line no-alert
                    }
                }
            });
        },
        setAbsoluteSceneNumbering(forAbsolute, sceneAbsolute) {
            const indexerName = $('#indexer-name').val();
            const seriesId = $('#series-id').val();

            if (sceneAbsolute === '') {
                sceneAbsolute = null;
            }

            $.getJSON('home/setSceneNumbering', {
                indexername: indexerName,
                seriesid: seriesId,
                forAbsolute,
                sceneAbsolute
            }, data => {
                // Set the values we get back
                if (data.sceneAbsolute === null) {
                    $('#sceneAbsolute_' + seriesId + '_' + forAbsolute).val('');
                } else {
                    $('#sceneAbsolute_' + seriesId + '_' + forAbsolute).val(data.sceneAbsolute);
                }

                if (!data.success) {
                    if (data.errorMessage) {
                        alert(data.errorMessage); // eslint-disable-line no-alert
                    } else {
                        alert('Update failed.'); // eslint-disable-line no-alert
                    }
                }
            });
        },
        setInputValidInvalid(valid, el) {
            if (valid) {
                $(el).css({
                    'background-color': '#90EE90', // Green
                    'color': '#FFF', // eslint-disable-line quote-props
                    'font-weight': 'bold'
                });
                return true;
            }
            $(el).css({
                'background-color': '#FF0000', // Red
                'color': '#FFF!important', // eslint-disable-line quote-props
                'font-weight': 'bold'
            });
            return false;
        },
        // @TODO: OMG: This is just a basic json, in future it should be based on the CRUD route.
        // Get the season exceptions and the xem season mappings.
        getSeasonSceneExceptions() {
            const { indexer, id } = this;

            if (!indexer || !id) {
                console.warn('Unable to get season scene exceptions: Unknown series identifier');
                return;
            }

            apiRoute.get('home/getSeasonSceneExceptions', {
                params: {
                    indexername: indexer,
                    seriesid: id
                }
            }).then(response => {
                this.setSeasonSceneExceptions(response.data);
            }).catch(error => {
                console.error('Error getting season scene exceptions', error);
            });
        },
        // Set the season exception based on using the get_xem_numbering_for_show() for animes if available in data.xemNumbering,
        // or else try to map using just the data.season_exceptions.
        setSeasonSceneExceptions(data) {
            $.each(data.seasonExceptions, (season, nameExceptions) => {
                let foundInXem = false;
                // Check if it is a season name exception, we don't handle the show name exceptions here
                if (season >= 0) {
                    // Loop through the xem mapping, and check if there is a xem_season, that needs to show the season name exception
                    $.each(data.xemNumbering, (indexerSeason, xemSeason) => {
                        if (xemSeason === parseInt(season, 10)) {
                            foundInXem = true;
                            $('<img>', {
                                id: 'xem-exception-season-' + xemSeason,
                                alt: '[xem]',
                                height: '16',
                                width: '16',
                                src: 'images/xem.png',
                                title: nameExceptions.join(', ')
                            }).appendTo('[data-season=' + indexerSeason + ']');
                        }
                    });

                    // This is not a xem season exception, let's set the exceptions as a medusa exception
                    if (!foundInXem) {
                        $('<img>', {
                            id: 'xem-exception-season-' + season,
                            alt: '[medusa]',
                            height: '16',
                            width: '16',
                            src: 'images/ico/favicon-16.png',
                            title: nameExceptions.join(', ')
                        }).appendTo('[data-season=' + season + ']');
                    }
                }
            });
        }
    }
};
</script>

<style>

</style>
