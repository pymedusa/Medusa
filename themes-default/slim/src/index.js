import $ from 'jquery';
import 'bootstrap'; // eslint-disable-line import/no-unassigned-import
import 'bootstrap/dist/css/bootstrap.min.css'; // eslint-disable-line import/no-unassigned-import
import './css/open-sans.css'; // eslint-disable-line import/no-unassigned-import

import Vue from 'vue';
import Vuex from 'vuex';
import VueMeta from 'vue-meta';
import VueRouter from 'vue-router';
import VueNativeSock from 'vue-native-websocket';
import AsyncComputed from 'vue-async-computed';
import ToggleButton from 'vue-js-toggle-button';
import Snotify from 'vue-snotify';
import axios from 'axios';
import debounce from 'lodash/debounce';
import store from './store';
import router from './router';
import { isDevelopment } from './utils';
import { apiRoute, apiv1, api, webRoot, apiKey } from './api';
import { Asset, Backstretch, DisplayShow, PlotInfo, ShowSelector, ScrollButtons, FileBrowser, NamePattern, AppHeader, AppLink, SelectList, LanguageSelect, RootDirs, AnidbReleaseGroupUi, SnatchSelection } from './components';

if (window) {
    window.isDevelopment = isDevelopment;

    // Adding libs to window so mako files can use them
    window.$ = $;
    window.jQuery = $;
    window.Vue = Vue;
    window.Vuex = Vuex;
    window.VueMeta = VueMeta;
    window.VueRouter = VueRouter;
    window.VueNativeSock = VueNativeSock;
    window.AsyncComputed = AsyncComputed;
    window.ToggleButton = ToggleButton;
    window.Snotify = Snotify;
    window.axios = axios;
    window._ = { debounce };
    window.store = store;
    window.router = router;
    window.apiRoute = apiRoute;
    window.apiv1 = apiv1;
    window.api = api;

    window.MEDUSA = {
        common: {},
        config: {},
        home: {},
        manage: {},
        history: {},
        errorlogs: {},
        schedule: {},
        addShows: {}
    };
    window.webRoot = webRoot;
    window.apiKey = apiKey;
    window.apiRoot = webRoot + '/api/v2/';

    // Push pages that load via a vue file but still use `el` for mounting
    window.components = [];
    window.components.push(AnidbReleaseGroupUi);
    window.components.push(AppHeader);
    window.components.push(AppLink);
    window.components.push(Asset);
    window.components.push(Backstretch);
    window.components.push(DisplayShow);
    window.components.push(FileBrowser);
    window.components.push(LanguageSelect);
    window.components.push(NamePattern);
    window.components.push(PlotInfo);
    window.components.push(RootDirs);
    window.components.push(ScrollButtons);
    window.components.push(SelectList);
    window.components.push(ShowSelector);
    window.components.push(SnatchSelection);
}
const UTIL = {
    exec(controller, action) {
        const ns = MEDUSA;
        action = (action === undefined) ? 'init' : action;

        if (controller !== '' && ns[controller] && typeof ns[controller][action] === 'function') {
            ns[controller][action]();
        }
    },
    init() {
        if (typeof startVue === 'function') { // eslint-disable-line no-undef
            startVue(); // eslint-disable-line no-undef
        } else {
            $('[v-cloak]').removeAttr('v-cloak');
        }

        const { body } = document;
        $('[asset]').each(function() {
            const asset = $(this).attr('asset');
            const series = $(this).attr('series');
            const path = apiRoot + 'series/' + series + '/asset/' + asset + '?api_key=' + apiKey;
            if (this.tagName.toLowerCase() === 'img') {
                const defaultPath = $(this).attr('src');
                if ($(this).attr('lazy') === 'on') {
                    $(this).attr('data-original', path);
                } else {
                    $(this).attr('src', path);
                }
                $(this).attr('onerror', 'this.src = "' + defaultPath + '"; return false;');
            }
            if (this.tagName.toLowerCase() === 'a') {
                $(this).attr('href', path);
            }
        });
        const controller = body.getAttribute('data-controller');
        const action = body.getAttribute('data-action');

        UTIL.exec('common'); // Load common
        UTIL.exec(controller); // Load MEDUSA[controller]
        UTIL.exec(controller, action); // Load MEDUSA[controller][action]

        window.dispatchEvent(new Event('medusa-loaded'));
    }
};

$.fn.extend({
    addRemoveWarningClass(_) {
        if (_) {
            return $(this).removeClass('warning');
        }
        return $(this).addClass('warning');
    }
});

const { pathname } = window.location;
if (!pathname.includes('/login') && !pathname.includes('/apibuilder')) {
    api.get('config/main').then(response => {
        $.extend(MEDUSA.config, response.data);
        MEDUSA.config.themeSpinner = MEDUSA.config.themeName === 'dark' ? '-dark' : '';
        MEDUSA.config.loading = '<img src="images/loading16' + MEDUSA.config.themeSpinner + '.gif" height="16" width="16" />';

        if (navigator.userAgent.indexOf('PhantomJS') === -1) {
            $(document).ready(UTIL.init);
        }

        MEDUSA.config.indexers.indexerIdToName = indexerId => {
            if (!indexerId) {
                return '';
            }
            return Object.keys(MEDUSA.config.indexers.config.indexers).filter(indexer => { // eslint-disable-line array-callback-return
                if (MEDUSA.config.indexers.config.indexers[indexer].id === parseInt(indexerId, 10)) {
                    return MEDUSA.config.indexers.config.indexers[indexer].name;
                }
            })[0];
        };

        MEDUSA.config.indexers.nameToIndexerId = name => {
            if (!name) {
                return '';
            }
            return MEDUSA.config.indexers.config.indexers[name];
        };
    }).catch(error => {
        console.debug(error);
        alert('Unable to connect to Medusa!'); // eslint-disable-line no-alert
    });
}
