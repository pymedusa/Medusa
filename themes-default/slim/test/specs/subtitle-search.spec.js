import Vuex, { Store } from 'vuex';
import VueRouter from 'vue-router';
import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import { SubtitleSearch } from '../../src/components';
import historyModule from '../../src/store/modules/history';
import show from '../__fixtures__/show-detailed';
import { result as subtitleResult } from '../__fixtures__/subtitle-search';
import episodeHistory from '../__fixtures__/episode-history';
import episode from '../__fixtures__/show-episode';

describe('SubtitleSearch', () => {
    let localVue;
    let store;
    const state = {
        history: {
            history: [],
            page: 0,
            showHistory: {},
            episodeHistory: {
                tvdb5692427: {
                    s04e06: episodeHistory
                }
            }
        }
    };

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuex);
        localVue.use(VueRouter);

        store = new Store({
            modules: {
                history: historyModule
            }
        });
        store.replaceState(state);
    });

    it('renders subtitle component with question', async () => {
        const wrapper = shallowMount(SubtitleSearch, {
            localVue,
            store,
            propsData: {
                show,
                episode
            }
        });

        wrapper.setData({
            displayQuestion: true
        });

        await wrapper.vm.$nextTick();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders subtitle component with manual subtitle results', async () => {
        const wrapper = mount(SubtitleSearch, {
            localVue,
            store,
            propsData: {
                show,
                episode
            }
        });

        wrapper.setData({
            subtitles: subtitleResult[0].subtitles,
            displayQuestion: false
        });

        await wrapper.vm.$nextTick();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders empty subtitle component', async () => {
        const wrapper = shallowMount(SubtitleSearch, {
            localVue,
            store,
            propsData: {
                show,
                episode
            }
        });

        wrapper.setData({
            displayQuestion: false
        });

        await wrapper.vm.$nextTick();
        expect(wrapper.element).toMatchSnapshot();
    });
});
