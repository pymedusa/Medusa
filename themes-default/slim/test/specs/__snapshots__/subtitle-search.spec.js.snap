// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubtitleSearch renders empty subtitle component 1`] = `
<tr
  class="subtitle-search-wrapper"
>
  <td
    colspan="9999"
  >
    <!---->
     
    <!---->
     
    <!---->
  </td>
</tr>
`;

exports[`SubtitleSearch renders subtitle component with manual subtitle results 1`] = `
<tr
  class="subtitle-search-wrapper"
>
  <td
    colspan="9999"
  >
    <!---->
     
    <!---->
     
    <!---->
  </td>
</tr>
`;

exports[`SubtitleSearch renders subtitle component with question 1`] = `
<tr
  class="subtitle-search-wrapper"
>
  <td
    colspan="9999"
  >
    <!---->
     
    <div
      class="search-question"
    >
      <div
        class="question"
      >
        <p>
          Do you want to manually pick subtitles or let us choose it for you?
        </p>
      </div>
       
      <div
        class="options"
      >
        <button
          class="btn-medusa btn-info"
          type="button"
        >
          Auto
        </button>
         
        <button
          class="btn-medusa btn-success"
          type="button"
        >
          Manual
        </button>
      </div>
    </div>
     
    <!---->
  </td>
</tr>
`;
